tools/autograd/gen_autograd_functions.py

```diff
@@ -191,11 +191,16 @@ def save_arg(arg, is_output):
     if uses_single_grad(func):
         body.append('auto& grad = grads[0];')
 
-    def emit_derivative(derivative):
+    def emit_derivative(derivative, forward_func_name):
         formula = derivative['formula']
         var_names = derivative['var_names']
         if len(var_names) == 1:
-            return DERIVATIVE_SINGLE.substitute(name=var_names[0], derivative=formula)
+            checks_any_grad_defined = False
+            if forward_func_name not in ['cat', 'stack'] and 'not_implemented' not in formula:
```

Comment:
Why are these two functions special cased here?

Reply:
`cat` and `stack`'s backward functions return a vector of Tensors, whereas all other backward functions that go through this codepath return a single Tensor. In the vector of Tensors case, the ternary `'any_grad_defined ? (' + formula + ') : Tensor()'` would not work.

Reply:
Would checking for the output type be ok here?
Otherwie, can you make that a nice list and document what it does (and mention it in the derivatives.yaml comment).

Reply:
Oh nevermind, in the caller, `func['args_with_derivatives']` has that information, so I should be able to do it.